pkg/ccl/telemetryccl/telemetryccl_test: TestTelemetry failed

pkg/ccl/telemetryccl/telemetryccl_test.TestTelemetry [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Ci_TestsAwsLinuxArm64_UnitTests/19463742?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Ci_TestsAwsLinuxArm64_UnitTests/19463742?buildTab=artifacts#/) on release-25.1 @ [913a508a9e1c8d5076767de40a97a6abbef08dc5](https://github.com/cockroachdb/cockroach/commits/913a508a9e1c8d5076767de40a97a6abbef08dc5):

Fatal error:

```
panic: test timed out after 5m0s
running tests:
	TestTelemetry (5m0s)
	TestTelemetry/multiregion_table (5s)
	TestTelemetry/multiregion_table/server (3s)

```
Stack: 

```
goroutine 296055 [running]:
testing.(*M).startAlarm.func1()
	GOROOT/src/testing/testing.go:2366 +0x30c
created by time.goFunc
	GOROOT/src/time/sleep.go:177 +0x38
```
<details><summary>Log preceding fatal error</summary>
<p>

```
        feature-usage [0 args]
        CREATE TABLE t1 ()
        ----
        sql.multiregion.create_table.locality.unspecified
    datadriven.go:144: 
        testdata/telemetry/multiregion_table:20:
        feature-usage [0 args]
        CREATE TABLE t2 () LOCALITY REGIONAL BY TABLE
        ----
        sql.multiregion.create_table.locality.regional_by_table
    datadriven.go:144: 
        testdata/telemetry/multiregion_table:25:
        feature-usage [0 args]
        CREATE TABLE t3 () LOCALITY REGIONAL BY TABLE IN "ap-southeast-2"
        ----
        sql.multiregion.create_table.locality.regional_by_table_in
    datadriven.go:144: 
        testdata/telemetry/multiregion_table:34:
        feature-usage [0 args]
        ALTER TABLE t1 SET LOCALITY REGIONAL BY ROW
        ----
        sql.multiregion.alter_table.locality.from.regional_by_table.to.regional_by_row
    datadriven.go:144: 
        testdata/telemetry/multiregion_table:39:
        exec [0 args]
        ALTER TABLE t1 SET LOCALITY REGIONAL BY TABLE
        ----
    datadriven.go:144: 
        testdata/telemetry/multiregion_table:43:
        feature-usage [0 args]
        ALTER TABLE t1 SET LOCALITY GLOBAL
        ----
        sql.multiregion.alter_table.locality.from.regional_by_table.to.global
    datadriven.go:144: 
        testdata/telemetry/multiregion_table:48:
        exec [0 args]
        ALTER TABLE t1 SET LOCALITY REGIONAL BY TABLE
        ----
    datadriven.go:144: 
        testdata/telemetry/multiregion_table:52:
        feature-usage [0 args]
        ALTER TABLE t1 SET LOCALITY REGIONAL BY TABLE IN "ap-southeast-2"
        ----
        sql.multiregion.alter_table.locality.from.regional_by_table.to.regional_by_table_in
    datadriven.go:144: 
        testdata/telemetry/multiregion_table:57:
        exec [0 args]
        ALTER TABLE t1 SET LOCALITY REGIONAL BY TABLE;
        ALTER TABLE t1 ADD COLUMN cr crdb_internal_region NOT NULL
        ----
```
</p>
</details>
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
/cc @cockroachdb/obs-prs
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestTelemetry.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/bazci/githubpost/issues)

</sub>


Jira issue: CRDB-49491

--- a/pkg/ccl/telemetryccl/testdata/telemetry/multiregion_table
+++ b/pkg/ccl/telemetryccl/testdata/telemetry/multiregion_table
@@ -12,117 +12,123 @@ USE d;
 ALTER DATABASE d ADD REGION "ap-southeast-2"
 ----
 
+#####################################################################
+# CREATE TABLE: Test initial table creation with different localities
+#####################################################################
+
 feature-usage
-CREATE TABLE t1 ()
+CREATE TABLE t0 () 
 ----
 sql.multiregion.create_table.locality.unspecified
 
 feature-usage
-CREATE TABLE t2 () LOCALITY REGIONAL BY TABLE
+CREATE TABLE t1 () LOCALITY REGIONAL BY TABLE
 ----
 sql.multiregion.create_table.locality.regional_by_table
 
 feature-usage
-CREATE TABLE t3 () LOCALITY REGIONAL BY TABLE IN "ap-southeast-2"
+CREATE TABLE t2 () LOCALITY REGIONAL BY TABLE IN "ap-southeast-2"
 ----
 sql.multiregion.create_table.locality.regional_by_table_in
 
-#
-# REGIONAL BY TABLE -> the others
-#
+feature-usage
+CREATE TABLE t3 () LOCALITY GLOBAL
+----
+sql.multiregion.create_table.locality.global
+
+#####################################################################
+# ALTER TABLE from REGIONAL BY TABLE
+#####################################################################
+
+exec
+CREATE TABLE t1_to_row () LOCALITY REGIONAL BY TABLE;
+CREATE TABLE t1_to_global () LOCALITY REGIONAL BY TABLE;
+CREATE TABLE t1_to_table_in () LOCALITY REGIONAL BY TABLE;
+CREATE TABLE t1_to_row_as () LOCALITY REGIONAL BY TABLE;
+ALTER TABLE t1_to_row_as ADD COLUMN cr crdb_internal_region NOT NULL;
+----
 
 feature-usage
-ALTER TABLE t1 SET LOCALITY REGIONAL BY ROW
+ALTER TABLE t1_to_row SET LOCALITY REGIONAL BY ROW
 ----
 sql.multiregion.alter_table.locality.from.regional_by_table.to.regional_by_row
 
-exec
-ALTER TABLE t1 SET LOCALITY REGIONAL BY TABLE
-----
-
 feature-usage
-ALTER TABLE t1 SET LOCALITY GLOBAL
+ALTER TABLE t1_to_global SET LOCALITY GLOBAL
 ----
 sql.multiregion.alter_table.locality.from.regional_by_table.to.global
 
-exec
-ALTER TABLE t1 SET LOCALITY REGIONAL BY TABLE
-----
-
 feature-usage
-ALTER TABLE t1 SET LOCALITY REGIONAL BY TABLE IN "ap-southeast-2"
+ALTER TABLE t1_to_table_in SET LOCALITY REGIONAL BY TABLE IN "ap-southeast-2"
 ----
 sql.multiregion.alter_table.locality.from.regional_by_table.to.regional_by_table_in
 
-exec
-ALTER TABLE t1 SET LOCALITY REGIONAL BY TABLE;
-ALTER TABLE t1 ADD COLUMN cr crdb_internal_region NOT NULL
-----
-
 feature-usage
-ALTER TABLE t1 SET LOCALITY REGIONAL BY ROW AS "cr"
+ALTER TABLE t1_to_row_as SET LOCALITY REGIONAL BY ROW AS "cr"
 ----
 sql.multiregion.alter_table.locality.from.regional_by_table.to.regional_by_row_as
 
+#####################################################################
+# ALTER TABLE from REGIONAL BY TABLE IN
+#####################################################################
+
 exec
-ALTER TABLE t1 SET LOCALITY REGIONAL BY TABLE
+CREATE TABLE t2_to_row () LOCALITY REGIONAL BY TABLE IN "ap-southeast-2";
+CREATE TABLE t2_to_global () LOCALITY REGIONAL BY TABLE IN "ap-southeast-2";
+CREATE TABLE t2_to_table () LOCALITY REGIONAL BY TABLE IN "ap-southeast-2";
+CREATE TABLE t2_to_row_as () LOCALITY REGIONAL BY TABLE IN "ap-southeast-2";
+ALTER TABLE t2_to_row_as ADD COLUMN cr crdb_internal_region NOT NULL;
 ----
 
 feature-usage
-ALTER TABLE t1 SET LOCALITY REGIONAL BY TABLE
-----
-sql.multiregion.alter_table.locality.from.regional_by_table.to.regional_by_table
-
-exec
-ALTER TABLE t1 SET LOCALITY REGIONAL BY TABLE
-----
-
-#
-# REGIONAL BY TABLE IN "ap-southeast-2" -> the others
-#
-
-feature-usage
-ALTER TABLE t3 SET LOCALITY REGIONAL BY ROW
+ALTER TABLE t2_to_row SET LOCALITY REGIONAL BY ROW
 ----
 sql.multiregion.alter_table.locality.from.regional_by_table_in.to.regional_by_row
 
-exec
-ALTER TABLE t3 SET LOCALITY REGIONAL BY TABLE IN "ap-southeast-2"
-----
-
 feature-usage
-ALTER TABLE t3 SET LOCALITY GLOBAL
+ALTER TABLE t2_to_global SET LOCALITY GLOBAL
 ----
 sql.multiregion.alter_table.locality.from.regional_by_table_in.to.global
 
-exec
-ALTER TABLE t3 SET LOCALITY REGIONAL BY TABLE IN "ap-southeast-2"
-----
-
 feature-usage
-ALTER TABLE t3 SET LOCALITY REGIONAL BY TABLE IN "ap-southeast-2"
-----
-sql.multiregion.alter_table.locality.from.regional_by_table_in.to.regional_by_table_in
-
-exec
-ALTER TABLE t3 SET LOCALITY REGIONAL BY TABLE IN "ap-southeast-2";
-ALTER TABLE t3 ADD COLUMN cr crdb_internal_region NOT NULL
-----
-
-feature-usage
-ALTER TABLE t3 SET LOCALITY REGIONAL BY ROW AS "cr"
-----
-sql.multiregion.alter_table.locality.from.regional_by_table_in.to.regional_by_row_as
-
-exec
-ALTER TABLE t3 SET LOCALITY REGIONAL BY TABLE IN "ap-southeast-2"
-----
-
-feature-usage
-ALTER TABLE t3 SET LOCALITY REGIONAL BY TABLE
+ALTER TABLE t2_to_table SET LOCALITY REGIONAL BY TABLE
 ----
 sql.multiregion.alter_table.locality.from.regional_by_table_in.to.regional_by_table
 
-exec
-ALTER TABLE t3 SET LOCALITY REGIONAL BY TABLE IN "ap-southeast-2"
+feature-usage
+ALTER TABLE t2_to_row_as SET LOCALITY REGIONAL BY ROW AS "cr"
 ----
+sql.multiregion.alter_table.locality.from.regional_by_table_in.to.regional_by_row_as
+
+
+#####################################################################
+# ALTER TABLE from GLOBAL
+#####################################################################exec
+
+exec
+CREATE TABLE t3_to_row () LOCALITY GLOBAL;
+CREATE TABLE t3_to_table () LOCALITY GLOBAL;
+CREATE TABLE t3_to_table_in () LOCALITY GLOBAL;
+CREATE TABLE t3_to_row_as () LOCALITY GLOBAL;
+ALTER TABLE t3_to_row_as ADD COLUMN cr crdb_internal_region NOT NULL;
+----
+
+feature-usage
+ALTER TABLE t3_to_row SET LOCALITY REGIONAL BY ROW
+----
+sql.multiregion.alter_table.locality.from.global.to.regional_by_row
+
+feature-usage
+ALTER TABLE t3_to_table SET LOCALITY REGIONAL BY TABLE
+----
+sql.multiregion.alter_table.locality.from.global.to.regional_by_table
+
+feature-usage
+ALTER TABLE t3_to_table_in SET LOCALITY REGIONAL BY TABLE IN "ap-southeast-2"
+----
+sql.multiregion.alter_table.locality.from.global.to.regional_by_table_in
+
+feature-usage
+ALTER TABLE t3_to_row_as SET LOCALITY REGIONAL BY ROW AS "cr"
+----
+sql.multiregion.alter_table.locality.from.global.to.regional_by_row_as

